File: app/src/main/java/pro/butovanton/mega/JSONPlaceHolderApi.java
package pro.butovanton.mega;


import java.util.List;

import retrofit2.Call;
import retrofit2.http.GET;
import retrofit2.http.Query;

public interface JSONPlaceHolderApi {

    @GET("test.php")
    public Call<List<MModel>> getModel ();

    @GET("test.php")
    public Call<List<ModelDetail>> getModelDetail(@Query("id") int id);
}

File: app/src/main/java/pro/butovanton/mega/JSONPlaceHolderApi.kt
package pro.butovanton.mega

import retrofit2.Call
import retrofit2.http.GET
import retrofit2.http.Query

interface JSONPlaceHolderApi {
    @get:GET("test.php")
    val model: Call<List<MModel?>?>?

    @GET("test.php")
    fun getModelDetail(@Query("id") id: Int): Call<List<ModelDetail?>?>?
}